strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/repository/group/GroupRepositoryArtifactExistenceChecker.java

```diff
@@ -65,7 +67,7 @@ public boolean artifactExistsInTheGroupRepositorySubTree(final Repository groupR
             else
             {
                 final LayoutProvider layoutProvider = layoutProviderRegistry.getProvider(subRepository.getLayout());
-                if (layoutProvider.containsPath(subRepository, artifactPath))
+                if (layoutProvider.containsPath(subRepository, RepositoryFiles.stringValue(repositoryPath)))
```

Comment:
@sbespalov : Perhaps the `RepositoryPath` class could have a `toString()` instead? It would be much simpler/cleaner to use. What do you think?

Reply:
hm... I am not sure about this, maybe we can leave this for the future and keep this as an option